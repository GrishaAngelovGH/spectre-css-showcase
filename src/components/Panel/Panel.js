import Avatar from "../Avatar"

const Panel = () => (
  <div className='column col-6 col-xs-12'>
    <div className='panel'>
      <div className='panel-header text-center'>
        <figure className='avatar avatar-lg'><Avatar initials='JS' /></figure>
        <div className='panel-title h5 mt-10'>John Smith</div>
        <div className='panel-subtitle'>221 B Baker Street</div>
      </div>
      <nav className='panel-nav'>
        <ul className='tab tab-block'>
          <li className='tab-item active'><a href='#panels'>Profile</a></li>
          <li className='tab-item'><a href='#panels'>Files</a></li>
          <li className='tab-item'><a href='#panels'>Tasks</a></li>
        </ul>
      </nav>
      <div className='panel-body'>
        <div className='tile tile-centered'>
          <div className='tile-content'>
            <div className='tile-title text-bold'>E-mail</div>
            <div className='tile-subtitle'>bruce.banner@hulk.com</div>
          </div>
          <div className='tile-action'>
            <button className='btn btn-link btn-action btn-lg tooltip tooltip-left' data-tooltip='Edit E-mail'>
              <i className='icon icon-edit'></i>
            </button>
          </div>
        </div>
        <div className='tile tile-centered'>
          <div className='tile-content'>
            <div className='tile-title text-bold'>Skype</div>
            <div className='tile-subtitle'>bruce.banner</div>
          </div>
          <div className='tile-action'>
            <button className='btn btn-link btn-action btn-lg'>
              <i className='icon icon-edit'></i>
            </button>
          </div>
        </div>
        <div className='tile tile-centered'>
          <div className='tile-content'>
            <div className='tile-title text-bold'>Location</div>
            <div className='tile-subtitle'>Dayton, Ohio</div>
          </div>
          <div className='tile-action'>
            <button className='btn btn-link btn-action btn-lg'>
              <i className='icon icon-edit'></i>
            </button>
          </div>
        </div>
      </div>
      <div className='panel-footer'>
        <button className='btn btn-primary btn-block'>Save</button>
      </div>
    </div>
  </div>
)

export default Panel