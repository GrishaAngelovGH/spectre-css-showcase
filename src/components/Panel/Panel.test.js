import { render } from '@testing-library/react'

import Panel from './Panel'

test('should render Panel component', () => {
  const view = render(<Panel />)

  expect(view).toMatchSnapshot()
})