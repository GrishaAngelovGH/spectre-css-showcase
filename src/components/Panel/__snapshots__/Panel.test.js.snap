// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Panel component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="column col-6 col-xs-12"
      >
        <div
          class="panel"
        >
          <div
            class="panel-header text-center"
          >
            <figure
              class="avatar avatar-lg"
            >
              <figure
                class="avatar avatar-sm mx-2"
                data-initial="JS"
                style="background: rgb(87, 85, 217);"
              />
            </figure>
            <div
              class="panel-title h5 mt-10"
            >
              John Smith
            </div>
            <div
              class="panel-subtitle"
            >
              221 B Baker Street
            </div>
          </div>
          <nav
            class="panel-nav"
          >
            <ul
              class="tab tab-block"
            >
              <li
                class="tab-item active"
              >
                <a
                  href="#panels"
                >
                  Profile
                </a>
              </li>
              <li
                class="tab-item"
              >
                <a
                  href="#panels"
                >
                  Files
                </a>
              </li>
              <li
                class="tab-item"
              >
                <a
                  href="#panels"
                >
                  Tasks
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="panel-body"
          >
            <div
              class="tile tile-centered"
            >
              <div
                class="tile-content"
              >
                <div
                  class="tile-title text-bold"
                >
                  E-mail
                </div>
                <div
                  class="tile-subtitle"
                >
                  bruce.banner@hulk.com
                </div>
              </div>
              <div
                class="tile-action"
              >
                <button
                  class="btn btn-link btn-action btn-lg tooltip tooltip-left"
                  data-tooltip="Edit E-mail"
                >
                  <i
                    class="icon icon-edit"
                  />
                </button>
              </div>
            </div>
            <div
              class="tile tile-centered"
            >
              <div
                class="tile-content"
              >
                <div
                  class="tile-title text-bold"
                >
                  Skype
                </div>
                <div
                  class="tile-subtitle"
                >
                  bruce.banner
                </div>
              </div>
              <div
                class="tile-action"
              >
                <button
                  class="btn btn-link btn-action btn-lg"
                >
                  <i
                    class="icon icon-edit"
                  />
                </button>
              </div>
            </div>
            <div
              class="tile tile-centered"
            >
              <div
                class="tile-content"
              >
                <div
                  class="tile-title text-bold"
                >
                  Location
                </div>
                <div
                  class="tile-subtitle"
                >
                  Dayton, Ohio
                </div>
              </div>
              <div
                class="tile-action"
              >
                <button
                  class="btn btn-link btn-action btn-lg"
                >
                  <i
                    class="icon icon-edit"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="panel-footer"
          >
            <button
              class="btn btn-primary btn-block"
            >
              Save
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="column col-6 col-xs-12"
    >
      <div
        class="panel"
      >
        <div
          class="panel-header text-center"
        >
          <figure
            class="avatar avatar-lg"
          >
            <figure
              class="avatar avatar-sm mx-2"
              data-initial="JS"
              style="background: rgb(87, 85, 217);"
            />
          </figure>
          <div
            class="panel-title h5 mt-10"
          >
            John Smith
          </div>
          <div
            class="panel-subtitle"
          >
            221 B Baker Street
          </div>
        </div>
        <nav
          class="panel-nav"
        >
          <ul
            class="tab tab-block"
          >
            <li
              class="tab-item active"
            >
              <a
                href="#panels"
              >
                Profile
              </a>
            </li>
            <li
              class="tab-item"
            >
              <a
                href="#panels"
              >
                Files
              </a>
            </li>
            <li
              class="tab-item"
            >
              <a
                href="#panels"
              >
                Tasks
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="panel-body"
        >
          <div
            class="tile tile-centered"
          >
            <div
              class="tile-content"
            >
              <div
                class="tile-title text-bold"
              >
                E-mail
              </div>
              <div
                class="tile-subtitle"
              >
                bruce.banner@hulk.com
              </div>
            </div>
            <div
              class="tile-action"
            >
              <button
                class="btn btn-link btn-action btn-lg tooltip tooltip-left"
                data-tooltip="Edit E-mail"
              >
                <i
                  class="icon icon-edit"
                />
              </button>
            </div>
          </div>
          <div
            class="tile tile-centered"
          >
            <div
              class="tile-content"
            >
              <div
                class="tile-title text-bold"
              >
                Skype
              </div>
              <div
                class="tile-subtitle"
              >
                bruce.banner
              </div>
            </div>
            <div
              class="tile-action"
            >
              <button
                class="btn btn-link btn-action btn-lg"
              >
                <i
                  class="icon icon-edit"
                />
              </button>
            </div>
          </div>
          <div
            class="tile tile-centered"
          >
            <div
              class="tile-content"
            >
              <div
                class="tile-title text-bold"
              >
                Location
              </div>
              <div
                class="tile-subtitle"
              >
                Dayton, Ohio
              </div>
            </div>
            <div
              class="tile-action"
            >
              <button
                class="btn btn-link btn-action btn-lg"
              >
                <i
                  class="icon icon-edit"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="panel-footer"
        >
          <button
            class="btn btn-primary btn-block"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
